plugins/package-managers/node/src/main/kotlin/utils/NpmSupport.kt

```diff
@@ -153,15 +154,22 @@ internal fun parseNpmVcsInfo(node: JsonNode): VcsInfo {
         revision = head
     )
 
-    val type = repository["type"].textValueOrEmpty()
-    val url = repository.textValue() ?: repository["url"].textValueOrEmpty()
+    if (repository.isTextual) {
+        val url = expandNpmShortcutUrl(repository.textValue())
+        val vcsInfo = VcsHost.parseUrl(url)
+        return vcsInfo.copy(revision = head.takeIf { it.isSha1() } ?: vcsInfo.revision.takeIf { it.isSha1() }.orEmpty())
```

Comment:
While I did read point 2. of your commit message, I do not understand what problem it solves. Even if there was "master" in the metadata, still a tag that matches the version would get precedence if I'm reading the code correctly:

https://github.com/oss-review-toolkit/ort/blob/6d04874140e4bf760ade6f1657173b67d0f4db41/downloader/src/main/kotlin/VersionControlSystem.kt#L346-L356

We first add our guessed revisions as candidates, and then any revisions from metadata. So how does it help to only take revisions if they are commit hashes, and map "master" to ""?

Reply:
I wasn't aware of this bit of the code, so my comment about the tag matching was wrong.

However, when you have "master" as revision and no release tags, then either
1. Re-scanning (scanner) has no cache hit which can lead to long execution times. 
2. If there is cache hit because provenance resolution stored, then you scan some arbitrary 
    old revisions of the branch without noticing it. The revision is determined by when it was first scanned.

Reply:
If a package provenance was resolved from a branch the resolution is always re-attempted:
https://github.com/oss-review-toolkit/ort/blob/e62e68b008ea13967cd16bef1404c5d1eab42622/scanner/src/main/kotlin/provenance/PackageProvenanceResolver.kt#L173-L190

I would agree that if resolving the provenance from a branch is an issue we should solve this globally and not on the package manager level. For example, we could introduce a configuration option to disable this and enforce curations in this case. Currently `allowMovingRevisions` is hardcoded to `true` when getting the revision candidates:
https://github.com/oss-review-toolkit/ort/blob/e62e68b008ea13967cd16bef1404c5d1eab42622/scanner/src/main/kotlin/provenance/PackageProvenanceResolver.kt#L208-L209